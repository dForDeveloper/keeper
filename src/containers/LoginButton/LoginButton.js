import React, { Component } from 'react';
import firebase from 'firebase/app';
import 'firebase/auth';
import googleicon from '../../images/googleicon.svg';
import PropTypes from 'prop-types';
import { fetchData, createOptions } from '../../utils/api';
import { setUser, setNotes } from '../../actions';
import { connect } from 'react-redux';

export class LoginButton extends Component {
  handleClick = async () => {
    const provider = new firebase.auth.GoogleAuthProvider();
    try{
      const googleData = await firebase.auth().signInWithPopup(provider);
      const { displayName, email, uid } = googleData.user;
      const user = { displayName, email, uid };
      const url = 'http://localhost:3001/api/v1/users';
      const options = createOptions('POST', user);
      const response = await fetchData(url, options);
      const { notes } = await response.json();
      this.props.setUser(user);
      this.props.setNotes(notes);
    } catch (error) {
      console.log('error', error);
    }
  }

  render() {
    return (
      <div>
        <button className='Login--button' onClick={this.handleClick}>
          <img src={googleicon} alt='Google Icon' className='Login--icon'/>
          <span className='Login--text'>Sign in with Google</span>
        </button>
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  setUser: (user) => dispatch(setUser(user)),
  setNotes: (notes) => dispatch(setNotes(notes))
});

export default connect(null, mapDispatchToProps)(LoginButton);

LoginButton.propTypes = {
  setUser: PropTypes.func,
  setNotes: PropTypes.func
}