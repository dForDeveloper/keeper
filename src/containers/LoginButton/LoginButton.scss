.Login--button {
  max-width: 220px;
  min-height: 40px;
  padding: 8px 16px;
  text-align: left;
  width: 100%;
  border: none;
  border-radius: 2px;
  height: 36px;
  margin: 0;
  min-width: 64px;
  outline: none;
  cursor: pointer;
  box-shadow: $light-shadow;
  margin-top: 15px; 
}

.Login--icon {
  height: 18px;
  width: 18px;
  vertical-align: middle;
}

.Login--text {
  font-size: 14px;
  font-weight: 500;
  padding-left: 16px;
  color: #757575
}