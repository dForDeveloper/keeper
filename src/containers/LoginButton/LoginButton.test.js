import React from 'react';
import { shallow } from 'enzyme';
import { LoginButton } from './LoginButton';

describe('LoginButton', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<LoginButton />);
    expect(wrapper).toMatchSnapshot();
  });
});