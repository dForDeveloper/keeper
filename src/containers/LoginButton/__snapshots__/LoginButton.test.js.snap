// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginButton should match the snapshot 1`] = `
<div>
  <button
    className="Login--button"
    onClick={[Function]}
  >
    <img
      alt="Google Icon"
      className="Login--icon"
      src="googleicon.svg"
    />
    <span
      className="Login--text"
    >
      Sign in with Google
    </span>
  </button>
</div>
`;
